perception/lidar_transfusion/include/lidar_transfusion/preprocess/preprocess_kernel.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIDAR_TRANSFUSION__PREPROCESS__PREPROCESS_KERNEL_HPP_
+#define LIDAR_TRANSFUSION__PREPROCESS__PREPROCESS_KERNEL_HPP_
+
+#include "lidar_transfusion/transfusion_config.hpp"
+#include "lidar_transfusion/utils.hpp"
+
+#include <cuda_runtime_api.h>
+
+namespace lidar_transfusion
+{
+
+class PreprocessCuda
```

Comment:
I do not think that this header is the right location for this class (it is meant for kernels). It probably belongs in its own header

Reply:
This class follows same approach as `postprocess_kernel.hpp`, imo it's right location or I don't see your point of view. If so, please extend your comment

Reply:
Imo if there is a reason and general principle of not using OOP in .cu files, let me know please. In that case I would rather refactor this code and `lidar_centerpoint` as well